docs/core_docs/docs/guides/langsmith_evaluation.md

```diff
@@ -0,0 +1,233 @@
+  'LangSmith is a platform developed by LangChain that provides tools for debugging, testing, evaluating, and monitoring Language Learning Models (LLMs) and intelligent agents. It helps developers move from prototyping to production by offering features such as trace logging, performance metrics, evaluation tools, and dataset management. LangSmith seamlessly integrates with LangChain, an open-source framework for building LLM applications. The platform allows developers to create and manage prompts, collaborate with teams, collect feedback, and monitor the performance of their LLM applications.'
+]
+```
+
+After setting up your environment, your agent traces should appear in the Projects section on the LangSmith app. Congratulations!
+
+If the agent is not effectively using the tools, evaluate it to establish a baseline.
+
+## Evaluate the Chain
+
+LangSmith allows you to test and evaluate your LLM applications. Follow these steps to benchmark your agent:
+
+### 1. Create a LangSmith dataset
+
+Use the LangSmith client to create a dataset with input questions and corresponding labels.
+
+For more information on datasets, including how to create them from CSVs or other files or how to create them in the platform, please refer to the [LangSmith documentation](https://docs.smith.langchain.com/).
+
+```typescript
+const outputs = [
+  "LangChain is an open-source framework for building applications using large language models. It is also the name of the company building LangSmith.",
+  "LangSmith is a unified platform for debugging, testing, and monitoring language model applications and agents powered by LangChain",
+  "July 18, 2023",
+  "The langsmith cookbook is a github repository containing detailed examples of how to use LangSmith to debug, evaluate, and monitor large language model-powered applications.",
+  "September 5, 2023",
+];
+const datasetName = `lcjs-qa-${unique_id}`;
+const dataset = await client.createDataset(datasetName);
+
+await Promise.all(
+  inputs.map(async (input, i) => {
+    await client.createExample(
+      { input },
+      { output: outputs[i] },
+      {
+        datasetId: dataset.id,
+      }
+    );
+  })
+);
+```
+
+### 2. Configure evaluation
+
+Manually comparing the results of chains in the UI is effective, but it can be time consuming.
+It can be helpful to use automated metrics and AI-assisted feedback to evaluate your component's performance.
+
+Below, we will create a custom run evaluator that logs a heuristic evaluation.
+
+```typescript
+import { RunEvalConfig } from "langchain/smith";
```

Comment:
We should also install `langchain` above, I've also not been installing `@langchain/core` explicitly since it'll always be part of `langchain`